receiver/splunkhecreceiver/receiver.go

```diff
@@ -295,18 +372,46 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, err)
 		return
 	}
+
 	consumerErr := r.logsConsumer.ConsumeLogs(ctx, ld)
 
 	_ = bodyReader.Close()
 
 	if consumerErr != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, consumerErr)
 	} else {
-		r.writeSuccessResponse(ctx, resp, ld.LogRecordCount())
+		if len(channelID) > 0 && r.ackExt != nil {
+			r.writeSuccessResponseWithAck(ctx, resp, ld.LogRecordCount(), channelID)
+		} else {
+			r.writeSuccessResponse(ctx, resp, ld.LogRecordCount(), okRespBody)
+		}
 		r.obsrecv.EndLogsOp(ctx, metadata.Type.String(), slLen, nil)
 	}
 }
 
+func (r *splunkReceiver) extractChannelHeader(req *http.Request) (string, bool) {
+	headers, ok := req.Header[splunk.HTTPSplunkChannelHeader]
+	if !ok {
+		return "", false
+	}
+	return headers[0], true
+
+}
+
+func (r *splunkReceiver) validateChannelHeader(channelID string) error {
+	if len(channelID) == 0 {
+		return errors.New(responseErrDataChannelMissing)
+	}
+
+	// check validity of channel
+	_, err := uuid.Parse(channelID)
```

Comment:
Is that specified anywhere?

Reply:
can you clarify what you mean? like what is the format of a channel?

Reply:
Yes, where is that defined? Is it important that it is a UUID string?

Reply:
Right. I am adding a comment